hooks/pre-commit

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+for FILE in $(git diff --cached --name-only)
+do
+	if [[ "$FILE" =~ src/[A-Za-z0-9\ \-]+*\.(c|h|cpp|cc)$ ]]; then
+		echo Autoformatting $FILE with clang-format
+		clang-format-11 -style=file -i -- $FILE
```

Comment:
On my system (Manjaro with clang 11 installed), there's no `clang-format-11` command available. `clang-format` is available, though.
Is there anyway to make this script portable across Linux distros?

Reply:
Parsing the output of `clang-format --version` could work to detect clang-format 11. Though maybe an environment variable would work better?

Reply:
Mhm I don't know... does that `clang-format-11` command exists on your system?

Reply:
@JF002 Yes, `clang-format` on my system is 10 not 11, that's why I used that binary. I'll try to figure out how to autodetect this. If you want to use it meanwhile you can install a modified version in your `.git` folder.